io/src/real_sense_2_grabber.cpp

```diff
@@ -0,0 +1,354 @@
+/*
+* Software License Agreement (BSD License)
+*
+*  Point Cloud Library (PCL) - www.pointclouds.org
+*  Copyright (c) 2010-2011, Willow Garage, Inc.
+*  Copyright (c) 2012-, Open Perception, Inc.
+*
+*  All rights reserved.
+*
+*  Redistribution and use in source and binary forms, with or without
+*  modification, are permitted provided that the following conditions
+*  are met:
+*
+*   * Redistributions of source code must retain the above copyright
+*     notice, this list of conditions and the following disclaimer.
+*   * Redistributions in binary form must reproduce the above
+*     copyright notice, this list of conditions and the following
+*     disclaimer in the documentation and/or other materials provided
+*     with the distribution.
+*   * Neither the name of the copyright holder(s) nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*
+*/
+
+#include <pcl/io/eigen.h>
+#include <pcl/io/eigen.h>
+#include <pcl/io/boost.h>
+#include <pcl/io/grabber.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+
+#include <librealsense2/rs.hpp>
+#include <pcl/io/real_sense_2_grabber.h>
+#include <pcl/common/time.h>
+
+namespace pcl
+{
+
+  RealSense2Grabber::RealSense2Grabber (const uint32_t serial_number)
+    : running_ (false)
+    , quit_ (false)
+    , serial_number_ (serial_number)
+    , signal_PointXYZ (nullptr)
+    , signal_PointXYZI (nullptr)
+    , signal_PointXYZRGB (nullptr)
+    , signal_PointXYZRGBA (nullptr)
+    , fps_ (0)
+    , device_width_ (424)
+    , device_height_ (240)
+    , target_fps_ (30)
```

Comment:
Why did you choose this resolution?
(The default resolution seems to be 1280*720.)

Reply:
In our testing on the in_hand_scanner we found setting the resolution to the lower end made the app far more responsive then at the higher resolution.  We felt it would be a good starting point for devs who then could gradually increase to get the best results.